### ISSUE TITLE
Strange y_xml bug

### ISSUE THREAD

**Issue by BitFros7y** [pawn-lang/YSI-Includes#82]
First i have to let you know that i suck at markup, so please no hard feelings.

XML File im trying to load (Standard MTA Map File - Its just spawns for peds)

```
<map edf:definitions="editor_main">
    <ped id="ped (1)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-282.92261" posY="1189.97266" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (2)" dimension="0" model="0" interior="0" rotZ="0" posX="-268.19211" posY="1124.2832" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (3)" dimension="0" model="0" interior="0" rotZ="0" posX="-284.3663" posY="1140.51282" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (4)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-302.22211" posY="1106.39478" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (5)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-245.828" posY="1107.20679" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (6)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-349.6051" posY="1117.76074" posZ="26.8479" rotX="0" rotY="0"></ped>
    <ped id="ped (7)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-324.98001" posY="1084.80957" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (8)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-240.3748" posY="1026.97192" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (9)" dimension="0" model="0" interior="0" rotZ="0" posX="-265.99271" posY="1009.72412" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (10)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-308.13809" posY="1011.14221" posZ="19.5938" rotX="0" rotY="0"></ped>
    <ped id="ped (11)" dimension="0" model="0" interior="0" rotZ="0" posX="-212.8891" posY="992.5766" posZ="19.5286" rotX="0" rotY="0"></ped>
    <ped id="ped (12)" dimension="0" model="0" interior="0" rotZ="0" posX="-172.5018" posY="991.08002" posZ="19.1865" rotX="0" rotY="0"></ped>
    <ped id="ped (13)" dimension="0" model="0" interior="0" rotZ="0" posX="-154.3817" posY="1022.51782" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (14)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-134.3627" posY="1050.00256" posZ="19.934" rotX="0" rotY="0"></ped>
    <ped id="ped (15)" dimension="0" model="0" interior="0" rotZ="0" posX="-155.2126" posY="1091.44824" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (16)" dimension="0" model="0" interior="0" rotZ="0" posX="-100.08" posY="1086.55725" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (17)" dimension="0" model="0" interior="0" rotZ="0" posX="-146.1398" posY="1127.58447" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (18)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-139.6671" posY="1155.70032" posZ="28.0349" rotX="0" rotY="0"></ped>
    <ped id="ped (19)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-109.0845" posY="1152.89954" posZ="19.6649" rotX="0" rotY="0"></ped>
    <ped id="ped (20)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-121.5916" posY="1204.3949" posZ="25.1515" rotX="0" rotY="0"></ped>
    <ped id="ped (21)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-238.7618" posY="1217.6759" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (22)" dimension="0" model="0" interior="0" rotZ="0" posX="-227.2907" posY="1152.84229" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (23)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-83.0388" posY="1217.70129" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (24)" dimension="0" model="0" interior="0" rotZ="0" posX="-4.0504" posY="1208.88281" posZ="19.3527" rotX="0" rotY="0"></ped>
    <ped id="ped (25)" dimension="0" model="0" interior="0" rotZ="0" posX="-45.2513" posY="1171.31653" posZ="19.5252" rotX="0" rotY="0"></ped>
    <ped id="ped (26)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-64.7892" posY="1100.69531" posZ="19.5938" rotX="0" rotY="0"></ped>
    <ped id="ped (27)" dimension="0" model="0" interior="0" rotZ="0" posX="-58.1307" posY="1140.89563" posZ="19.7349" rotX="0" rotY="0"></ped>
    <ped id="ped (28)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-16.8674" posY="1097.81494" posZ="19.5938" rotX="0" rotY="0"></ped>
    <ped id="ped (29)" dimension="0" model="0" interior="0" rotZ="0" posX="29.6427" posY="1116.73413" posZ="19.5938" rotX="0" rotY="0"></ped>
    <ped id="ped (30)" dimension="0" model="0" interior="0" rotZ="0" posX="24.842" posY="1146.10315" posZ="19.5938" rotX="0" rotY="0"></ped>
    <ped id="ped (31)" dimension="0" model="0" interior="0" rotZ="0.003" posX="85.6302" posY="1184.20715" posZ="18.6641" rotX="0" rotY="0"></ped>
    <ped id="ped (32)" dimension="0" model="0" interior="0" rotZ="0" posX="38.9148" posY="1201.09387" posZ="18.8594" rotX="0" rotY="0"></ped>
    <ped id="ped (33)" dimension="0" model="0" interior="0" rotZ="0" posX="157.6559" posY="1172.62341" posZ="15.2419" rotX="0" rotY="0"></ped>
    <ped id="ped (34)" dimension="0" model="0" interior="0" rotZ="0" posX="-57.238" posY="1031.90015" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (35)" dimension="0" model="0" interior="0" rotZ="0" posX="-14.6783" posY="1018.638" posZ="19.7422" rotX="0" rotY="0"></ped>
    <ped id="ped (36)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-28.899" posY="949.97607" posZ="19.1548" rotX="0" rotY="0"></ped>
    <ped id="ped (37)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-69.0619" posY="936.9657" posZ="20.3626" rotX="0" rotY="0"></ped>
    <ped id="ped (38)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-63.9107" posY="885.78308" posZ="20.8349" rotX="0" rotY="0"></ped>
    <ped id="ped (39)" dimension="0" model="0" interior="0" rotZ="0" posX="-138.5905" posY="897.85449" posZ="18.6912" rotX="0" rotY="0"></ped>
    <ped id="ped (40)" dimension="0" model="0" interior="0" rotZ="0" posX="-140.2838" posY="867.27832" posZ="17.9861" rotX="0" rotY="0"></ped>
    <ped id="ped (41)" dimension="0" model="0" interior="0" rotZ="0" posX="-163.35201" posY="882.888" posZ="17.7337" rotX="0" rotY="0"></ped>
    <ped id="ped (42)" dimension="0" model="0" interior="0" rotZ="0" posX="-169.1898" posY="929.59192" posZ="17.649" rotX="0" rotY="0"></ped>
    <ped id="ped (43)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-256.48669" posY="947.01147" posZ="16.9738" rotX="0" rotY="0"></ped>
    <ped id="ped (44)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-238.5193" posY="901.33142" posZ="11.6841" rotX="0" rotY="0"></ped>
    <ped id="ped (45)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-270.09061" posY="906.85028" posZ="12.5378" rotX="0" rotY="0"></ped>
    <ped id="ped (46)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-200.54401" posY="939.36792" posZ="14.7277" rotX="0" rotY="0"></ped>
    <ped id="ped (47)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-326.8851" posY="957.15253" posZ="17.3872" rotX="0" rotY="0"></ped>
    <ped id="ped (48)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-366.7005" posY="940.11292" posZ="9.9341" rotX="0" rotY="0"></ped>
    <ped id="ped (49)" dimension="0" model="0" interior="0" rotZ="0" posX="-339.17041" posY="881.40112" posZ="8.8586" rotX="0" rotY="0"></ped>
    <ped id="ped (50)" dimension="0" model="0" interior="0" rotZ="0" posX="-397.43161" posY="899.68298" posZ="5.8074" rotX="0" rotY="0"></ped>
    <ped id="ped (51)" dimension="0" model="0" interior="0" rotZ="0" posX="-362.60541" posY="849.63959" posZ="11.5498" rotX="0" rotY="0"></ped>
    <ped id="ped (52)" dimension="0" model="0" interior="0" rotZ="0" posX="-400.81219" posY="823.97852" posZ="10.0049" rotX="0" rotY="0"></ped>
    <ped id="ped (53)" dimension="0" model="0" interior="0" rotZ="0" posX="-335.51401" posY="827.651" posZ="14.2422" rotX="0" rotY="0"></ped>
    <ped id="ped (54)" dimension="0" model="0" interior="0" rotZ="0" posX="-331.1879" posY="794.1814" posZ="17.1028" rotX="0" rotY="0"></ped>
    <ped id="ped (55)" dimension="0" model="0" interior="0" rotZ="0" posX="-266.466" posY="780.6283" posZ="16.5671" rotX="0" rotY="0"></ped>
    <ped id="ped (56)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-257.2536" posY="845.94342" posZ="12.8567" rotX="0" rotY="0"></ped>
    <ped id="ped (57)" dimension="0" model="0" interior="0" rotZ="0.003" posX="-230.74969" posY="840.70111" posZ="12.3819" rotX="0" rotY="0"></ped>
</map>
```

Code to load file (used example from wiki):

```
public OnFilterScriptInit()
{
    AntiDeAMX();
    print("\n--------------------------------------");
    print(" Blank Filterscript by your name here");
    print("--------------------------------------\n");

    xRules = XML_New();
    if (xRules != NO_XML_FILE)
    {
        // Add a function to handle the <ped> tag
        XML_AddHandler(xRules, "ped", "LoadSpawns");
        XML_Parse(xRules, "Spawns.map");
        //XML_ParseTag(xRules, "ped", 5);
    }
    return 1;
}

forward LoadSpawns();
public LoadSpawns()
{

    static
        key[MAX_XML_ENTRY_NAME],
        val[MAX_XML_ENTRY_TEXT];

    while (XML_GetKeyValue(key, val))
    {
        if( strlen(key) > 0 && strcmp(key, "id", true) == 0)
        {
            printf("Importing spawn: %s | %s", key, val);
        }
    }

    return 1;
}
```

Server log:

```
[01:12:34] Importing spawn: id | ped (2)
[01:12:34] Importing spawn: id | ped (3)
[01:12:34] Importing spawn: id | ped (5)
[01:12:34] Importing spawn: id | ped (9)
[01:12:34] Importing spawn: id | ped (10)
[01:12:34] Importing spawn: id | ped (11)
[01:12:34] Importing spawn: id | ped (12)
[01:12:34] Importing spawn: id | ped (13)
[01:12:34] Importing spawn: id | ped (15)
[01:12:34] Importing spawn: id | ped (16)
[01:12:34] Importing spawn: id | ped (17)
[01:12:34] Importing spawn: id | ped (22)
[01:12:34] Importing spawn: id | ped (24)
[01:12:34] Importing spawn: id | ped (25)
[01:12:34] Importing spawn: id | ped (27)
[01:12:34] Importing spawn: id | ped (29)
[01:12:34] Importing spawn: id | ped (30)
[01:12:34] Importing spawn: id | ped (31)
[01:12:34] Importing spawn: id | ped (32)
[01:12:34] Importing spawn: id | ped (33)
[01:12:34] Importing spawn: id | ped (34)
[01:12:34] Importing spawn: id | ped (35)
[01:12:34] Importing spawn: id | ped (36)
[01:12:34] Importing spawn: id | ped (37)
[01:12:34] Importing spawn: id | ped (38)
[01:12:34] Importing spawn: id | ped (39)
[01:12:34] Importing spawn: id | ped (40)
[01:12:34] Importing spawn: id | ped (41)
[01:12:34] Importing spawn: id | ped (42)
[01:12:34] Importing spawn: id | ped (48)
[01:12:34] Importing spawn: id | ped (49)
[01:12:34] Importing spawn: id | ped (50)
[01:12:34] Importing spawn: id | ped (51)
[01:12:34] Importing spawn: id | ped (52)
[01:12:34] Importing spawn: id | ped (53)
[01:12:34] Importing spawn: id | ped (54)
[01:12:34] Importing spawn: id | ped (55)
```

As you can see from the start, 1, 4, 6, 7, 8 are missing. In total it is loading about 36 or 37 lines out of 57 that there is. So for some unknown reason to me, some of the lines are failing to load...

**Comment by Y-Less** [comment]
Figured out why: the lines are too long.  y_xml was originally designed for highly nested files.  I'm actually slightly surprised this even worked at all.  You can either try and increase `YSI_MAX_STRING`, but that's not a simple single definition (because it is used a lot in assembly too), or put some linebreaks in your file.  I'll try get a proper fix.

**Comment by Y-Less** [comment]
Fixed in #101.
